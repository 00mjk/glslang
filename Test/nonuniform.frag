#version 450

nonuniformEXT in vec4 nu_inv4;
nonuniformEXT float nu_gf;

nonuniformEXT out vec4 nu_outv4;           // ERROR, out
nonuniformEXT uniform vec4 nu_uv4;         // ERROR, uniform
nonuniformEXT const float nu_constf = 1.0; // ERROR, const

void foo(nonuniformEXT int nupi)
{
}

void main()
{
    nonuniformEXT int nu_li;
    nonuniformEXT const int nu_ci = 2; // ERROR, const

    foo(nu_li);

    int a;
    nu_li = nonuniformEXT(a) + nonuniformEXT(a * 2);
    nu_li = nonuniformEXT(a, a);       // ERROR, too many arguments
    nu_li = nonuniformEXT();           // ERROR, no arguments
}