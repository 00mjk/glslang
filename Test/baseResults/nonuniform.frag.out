nonuniform.frag
ERROR: 0:6: 'nonuniformEXT' : for non-parameter, can only apply to 'in' or no storage qualifier 
ERROR: 0:7: 'nonuniformEXT' : for non-parameter, can only apply to 'in' or no storage qualifier 
ERROR: 0:8: 'nonuniformEXT' : for non-parameter, can only apply to 'in' or no storage qualifier 
ERROR: 0:18: 'nonuniformEXT' : for non-parameter, can only apply to 'in' or no storage qualifier 
ERROR: 0:24: 'constructor' : too many arguments 
ERROR: 0:24: 'assign' :  cannot convert from ' const float' to ' nonuniform temp int'
ERROR: 0:25: 'constructor' : not enough data provided for construction 
ERROR: 0:25: 'assign' :  cannot convert from ' const float' to ' nonuniform temp int'
ERROR: 0:28: 'nonuniformEXT' : not allowed on block or structure members 
ERROR: 0:29: 'nonuniformEXT' : not allowed on block or structure members 
ERROR: 10 compilation errors.  No code generated.


Shader version: 450
ERROR: node is still EOpNull!
0:10  Function Definition: foo(i1;i1; ( nonuniform temp int)
0:10    Function Parameters: 
0:10      'nupi' ( nonuniform in int)
0:10      'f' ( nonuniform out int)
0:12    Sequence
0:12      Branch: Return with expression
0:12        'nupi' ( nonuniform in int)
0:15  Function Definition: main( ( global void)
0:15    Function Parameters: 
0:?     Sequence
0:20      Function Call: foo(i1;i1; ( nonuniform temp int)
0:20        'nu_li' ( nonuniform temp int)
0:20        'nu_li' ( nonuniform temp int)
0:23      move second child to first child ( temp int)
0:23        'nu_li' ( nonuniform temp int)
0:23        add ( temp int)
0:23          'a' ( nonuniform temp int)
0:23          component-wise multiply ( nonuniform temp int)
0:23            'a' ( temp int)
0:23            Constant:
0:23              2 (const int)
0:24      'nu_li' ( nonuniform temp int)
0:25      'nu_li' ( nonuniform temp int)
0:?   Linker Objects
0:?     'nu_inv4' ( smooth nonuniform in 4-component vector of float)
0:?     'nu_gf' ( nonuniform temp float)
0:?     'nu_outv4' ( nonuniform out 4-component vector of float)
0:?     'nu_uv4' ( nonuniform uniform 4-component vector of float)
0:?     'nu_constf' ( nonuniform const float)
0:?       1.000000
0:?     'ins' (layout( location=1) smooth in structure{ global float a,  temp float b})
0:?     'inb' (layout( location=3) in block{ in float a,  in float b})


Linked fragment stage:


Shader version: 450
ERROR: node is still EOpNull!
0:10  Function Definition: foo(i1;i1; ( nonuniform temp int)
0:10    Function Parameters: 
0:10      'nupi' ( nonuniform in int)
0:10      'f' ( nonuniform out int)
0:12    Sequence
0:12      Branch: Return with expression
0:12        'nupi' ( nonuniform in int)
0:15  Function Definition: main( ( global void)
0:15    Function Parameters: 
0:?     Sequence
0:20      Function Call: foo(i1;i1; ( nonuniform temp int)
0:20        'nu_li' ( nonuniform temp int)
0:20        'nu_li' ( nonuniform temp int)
0:23      move second child to first child ( temp int)
0:23        'nu_li' ( nonuniform temp int)
0:23        add ( temp int)
0:23          'a' ( nonuniform temp int)
0:23          component-wise multiply ( nonuniform temp int)
0:23            'a' ( temp int)
0:23            Constant:
0:23              2 (const int)
0:24      'nu_li' ( nonuniform temp int)
0:25      'nu_li' ( nonuniform temp int)
0:?   Linker Objects
0:?     'nu_inv4' ( smooth nonuniform in 4-component vector of float)
0:?     'nu_gf' ( nonuniform temp float)
0:?     'nu_outv4' ( nonuniform out 4-component vector of float)
0:?     'nu_uv4' ( nonuniform uniform 4-component vector of float)
0:?     'nu_constf' ( nonuniform const float)
0:?       1.000000
0:?     'ins' (layout( location=1) smooth in structure{ global float a,  temp float b})
0:?     'inb' (layout( location=3) in block{ in float a,  in float b})

